federationexample/README.md

```diff
@@ -0,0 +1,135 @@
+# Graphql Federation
+
+In our current world, we have a single graphql schema that contains all theh obejcts and field funcs that need to run. However, as the number of objects and field funcs scale, we want to allow product teams to each have their own gqlserver that conatain the portion of the gqlschema relevant to the product team.
+
+Federation helps by creating a service called the gqlgateway that is aware of every team gqlserver and it's schemas. Queries are now sent to the gateway whichh parses and sends subqueries to each of the gqlservers. The subqueries only contains field funcs that the team's gqlserver know's how to resolve. After sending all the subqueries, it stitches the results into a single result sent back to the client. This ensures the client can query any field while the schema is federated under the hood. 
+
+We want to support this to help in terms of giving teams more ownership over their portion of the graphql schema along and with giving team's isolation such thhat there work with their portion of the graphhql schema doesn't effect other teams.
+
+## Architecture Diagram
+
+![Screenshot](federation.png)
+
+## API 
+
+If an object entirely belongs to a gqlserver, it can exist on only that gqlserver with all it's field funcs. However, there are some objects that have fields that belong to different gqlservers. We call these federated objects. 
+
+A nonfederated object looks something like this example below 
+```
+type Device struct {
+    Id            int64
+    OrgId         int64
+    Name          string
+}
+device := schema.Object("Device", Device{})
+```
+
+A federated object has an extra field called FetchObjectFromKeys that takes a function, telling the gqlserver how to fecth the full object from a the federated keys, the minimum number of fields needed to fetch the object. A federated object can exist on multiple gqlservers and hhave different field funcs on each of the team's gqlservers. 
+
+```
+type Device struct {
+    Id            int64
+    OrgId         int64
+    Name          string
+}
+// Device keys has to be a subset of the fields on Device and is the minimum information needded to fetch a full Device
+type DeviceKeys struct {
+    Id            int64
+    OrgId         int64
+}
+device := schema.Object("Device", Device{}, schemabuilder.FetchObjectFromKeys(func(args struct{ Keys []*DeviceKeys }) []*Device {
+    // Fetch device from database using device keys    
+    return fetchedDevices
+}))
+```
```

Comment:
This example is a bit confusing to me... can we work on this together to provide a bit more descriptive and detailed example of `federated object`?

Reply:
Sounds good! Let's find some time to talk about it tomorrow